connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -668,10 +669,16 @@ class SparkConnectPlanner(session: SparkSession) {
             case other => throw InvalidPlanInput(s"Invalid schema $other")
           }
         }
-        reader.load().queryExecution.analyzed
+        if (rel.getDataSource.getPathsCount == 0) {
+          reader.load().queryExecution.analyzed
+        } else if (rel.getDataSource.getPathsCount == 1) {
+          reader.load(rel.getDataSource.getPaths(0)).queryExecution.analyzed
+        } else {
+          reader.load(rel.getDataSource.getPathsList.asScala: _*).queryExecution.analyzed
+        }
```

Comment:
Can we just call:

```scala
reader.load(rel.getDataSource.getPathsList.asScala: _*).queryExecution.analyzed
```

alone without checking the length?

Reply:
good question. I am not very sure that the logics of single path and multi pathes are the same 

https://github.com/apache/spark/blob/595f14881f92a09b03888dd69f69f9e7edf2a060/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L175-L212

Reply:
👌